app/pages/sections/additional-services/order-item/controller.js

```diff
@@ -71,105 +64,30 @@ export const getOrderItemErrorPageContext = (params) => {
   return getErrorContext(updatedParams);
 };
 
-const formatPostData = ({
-  selectedRecipientId,
-  serviceRecipientName,
-  selectedSolutionId,
-  solutionName,
-  selectedPrice,
-  formData,
-}) => ({
-  ...selectedPrice,
-  serviceRecipient: {
-    name: serviceRecipientName,
-    odsCode: selectedRecipientId,
-  },
-  catalogueSolutionId: selectedSolutionId,
-  catalogueSolutionName: solutionName,
-  deliveryDate: extractDate('deliveryDate', formData),
-  quantity: parseInt(formData.quantity, 10),
-  estimationPeriod: formData.selectEstimationPeriod,
-  price: parseFloat(formData.price),
-});
-
-const postSolutionOrderItem = async ({
-  accessToken,
-  orderId,
-  selectedRecipientId,
-  serviceRecipientName,
-  selectedSolutionId,
-  solutionName,
-  selectedPrice,
-  formData,
-}) => {
-  const endpoint = getEndpoint({ api: 'ordapi', endpointLocator: 'postSolutionOrderItem', options: { orderId } });
-  const body = formatPostData({
-    selectedRecipientId,
-    serviceRecipientName,
-    selectedSolutionId,
-    solutionName,
-    selectedPrice,
-    formData,
-  });
-
-  await postData({
-    endpoint, body, accessToken, logger,
-  });
-  logger.info(`Order item for ${solutionName} and ${serviceRecipientName} successfully created for order id: ${orderId}`);
-  return { success: true };
-};
-
-const formatPutData = ({
-  formData,
-}) => ({
-  deliveryDate: extractDate('deliveryDate', formData),
-  quantity: parseInt(formData.quantity, 10),
-  estimationPeriod: formData.selectEstimationPeriod,
-  price: parseFloat(formData.price),
-});
-
-const putSolutionOrderItem = async ({
-  accessToken,
-  orderId,
-  orderItemId,
-  formData,
-}) => {
-  const endpoint = getEndpoint({ api: 'ordapi', endpointLocator: 'putCatalogueOrderItem', options: { orderId, orderItemId } });
-  const body = formatPutData({
-    formData,
-  });
-
-  await putData({
-    endpoint, body, accessToken, logger,
-  });
-  logger.info(`Order item successfully updated for order id: ${orderId} and order item id: ${orderItemId}`);
-  return { success: true };
-};
-
-export const saveSolutionOrderItem = async ({
+export const saveOrderItem = async ({
```

Comment:
Wow much neater!

Reply:
I can't wait to get my hands on the version of this in catalogue-solutions and retrofit it.